clang/test/AST/ByteCode/unions.cpp

```diff
@@ -504,4 +504,39 @@ namespace AnonymousUnion {
   static_assert(return_init_all().a.p == 7); // both-error {{}} \
                                              // both-note {{read of member 'p' of union with no active member}}
 }
+
+namespace InactiveDestroy {
+  struct A {
+    constexpr ~A() {}
+  };
+  union U {
+    A a;
+    constexpr ~U() {
```

Comment:
We should also add a test to check cases where inside `~U` we have `a.~A();` and we attempt to double destruct.